core/src/test/java/io/grpc/StaticTestingClassLoader.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2017, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc;
+
+import com.google.common.base.Preconditions;
+import io.grpc.internal.IoUtils;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.regex.Pattern;
+
+/**
+ * A class loader that can be used to repeatedly trigger static initialization of a class. A new
```

Comment:
"initialization" could simply mean what happens during creating a class instance: new Object() initializes an Object. So I'm really referring to "class initialization", but the only way that is really useful to test code is to trigger static code blocks and static field initializers to run. This can also be used if you are mutating static state after initialization. So that's why "Static" is important, as it describes its use-case.

I honestly tried pretty hard to come up with another name, but came up short. I mainly wanted to name it based on what it did, and only secondly based on how it was used. But I found it pretty hard. "TestingClassLoader" is a very weak name in my mind, since it would apply equally well to [ReplacingClassLoader](https://github.com/grpc/grpc-java/blob/v1.5.0/core/src/test/java/io/grpc/ReplacingClassLoader.java) (wow, it's Javadoc is bad...). I considered things like "RedefiningClassLoader" and "CopyingClassLoader" and more. And then I gave up and moved on with life.